api-report/tree2.api.md

```diff
@@ -1331,6 +1324,23 @@ export interface NodeExistsConstraint {
     violated: boolean;
 }
 
+// @alpha
+export type NodeIdentifier = Brand<StableId, "Node Identifier">;
+
+// @alpha (undocumented)
+export interface NodeIdentifierFieldKind extends BrandedFieldKind<"NodeIdentifier", Multiplicity.Value, FieldEditor<any>> {
+}
+
+// @alpha
+export const nodeIdentifierKey: GlobalFieldKey;
+
+// @alpha
+export function nodeIdentifierSchema(key: string): {
```

Comment:
for the near term, we expect to only have a single hard coded index for our main identifier field, right? Thus at the package export level, I think it makes more sense to export the field and its library (maybe from a file in the domains directory) rather than this builder.

Reply:
That's a great idea. I put it in domains (we have more than one domain now, yay!)